### PR TITLE
Add missing functions to Fortran use_mpi interface

### PR DESCRIPTION
**Pull request by N00byKing**
There's currently two closely related bugs.

The PMPI define for `MPI_Status_set_elements` is defined in `pmpi-ignore-tkr-interfaces.h` but the actual interface is in `mpi-ignore-tkr-interfaces-generated.h`.
Then, in `mpi-ignore-tkr.F90` both are included in that order. This leads to the preprocessor overwriting `MPI_Status_set_elements` with the PMPI version, and the normal one is omitted entirely.
This is fixed by the first commit which just removes the superfluous define.

Then, for the functions in `mpi-ignore-tkr-interfaces-generated.h`, no PMPI interface versions are being generated at the moment, since the define in that header is unused.
In the second commit, I include that header again in `mpi-ignore-tkr.F90` with the define to ensure the PMPI interfaces are in the final output.

I'm a bit unsure about the second commit, would be grateful for a deeper look.

### CODE REVIEW

**Comment by hppritcha** [comment]
You should only need the first commit to fix this problem.  The correct interfaces for both PMPI and MPI should be defined in the generated mpi-ignore-tkr-interfaces-generated.h.

**Comment by N00byKing** [comment]
mpi-ignore-tkr-interfaces-generated.h only creates a list of defines, guarded by OMPI_BUILD_MPI_PROFILING, for MPI->PMPI, not any new interfaces.
But as far as I can tell OMPI_BUILD_MPI_PROFILING is unused for use-mpi-ignore-tkr, and so without that commit none of these functions are getting a PMPI interface (The commit might still be the wrong fix though).

As for the build failure, it looks like there's something else broken with the _FTS suffixed functions. I'm taking a look now.

**Comment by N00byKing** [comment]
So it looks like I stumbled into a rabbit hole...

With the second commit of this PR I added interfaces for the PMPI versions of all generated use-mpi-ignore-tkr functions,
including `MPI_Comm_attach_buffer_FTS` and `MPI_Session_attach_buffer_FTS`.
However, the python `print_profiling_rename_macros` function is broken for f90.
It currently adds a define like this:
```
#define MPI_Comm_attach_buffer_FTS PMPI_Comm_attach_buffer_FTS
```
But the interface does not have the _FTS suffix, which leads to:
```
interface MPI_Session_attach_buffer
subroutine PMPI_Session_attach_buffer_FTS(session, &
```
Which is incorrect, and leads to multiple equivalent interface names, one for (P)MPI each but with the same interface name.

Correct example from f08:
```
#define MPI_Accumulate_f08ts PMPI_Accumulate_f08ts
#define MPI_Accumulate_c_f08ts PMPI_Accumulate_c_f08ts
#define MPI_Accumulate PMPI_Accumulate
```
Meaning a define for the TS, big, and interface name each.

The reason for that is because its gated away in `fortran.py` explicitly:
```
if gen_f90 == False:
    name = util.fortran_f08_generic_interface_name(template.prototype.name)
    out.dump(f'#define {name} P{name}')
```
I'm pretty sure this is to get the non _f08 suffixed name for f08, but it ignores the _FTS suffix for f90.

I've added a commit ([a76e35e](https://github.com/open-mpi/ompi/pull/14219/commits/a76e35e24b7c79713b9fcfdea4a9ec452ef8ed6d)) which fixes this by collecting all required names (TS, big and interface if interface != TS),
then emits the define for each.

But when testing this, I got another error.
It turns out the two _FTS functions are not being compiled, and prior to this PR they seem to already be broken and cause linker errors.
They need a generated c source similar to how f08 does it (calling bindings.py with `code` instead of `interface`), which is currently not being created.
So on `main` using these two functions will give you a linker error.

I've essentially recreated the f08 setup in f90 in commit [2b711dc](https://github.com/open-mpi/ompi/pull/14219/commits/2b711dcc838bfc91007885bb4ab8755e64a5190e) in order to get this to work.

Finally, [b6a6731](https://github.com/open-mpi/ompi/pull/14219/commits/b6a6731bb62341ecf327cd67f9fc207243cc2ba1) just makes sure that the generated interface is recreated if the script is modified, which tripped me up a few times.

**Comment by hppritcha** [comment]
@devreal could you do a co-pilot scan on this PR?

**Comment by hppritcha** [comment]
This PR looks good to me modulo the copilot finding above and something I noticed in examining the generated api_f90_generated.F90.  Namely the 'x' arguments i think should really be replaced by the proper argument name in the standard to allow for keyword argument asociation.  Fixing this would involve updating line 363 in fortran.py, but the comments above that line give me pause.  See 86c5eab768534b9bac5b7e3b6e6e9650ab45637b .  i'm not sure that comment is right for use mpi.

File: ompi/mpi/bindings/ompi_bindings/fortran.py [hunk 1]
```diff
@@ -292,14 +296,14 @@ def print_profiling_rename_macros(templates, out, args):
...
-        name = util.fortran_name(template.prototype.name, gen_f90=gen_f90, needs_ts=needs_ts)
-        out.dump(f'#define {name} P{name}')
+        names = [util.fortran_name(template.prototype.name, gen_f90=gen_f90, needs_ts=needs_ts)]
         # Check for bigcount version
         if util.prototype_has_bigcount(template.prototype):
-            bigcount_name = util.fortran_name(template.prototype.name, bigcount=True, needs_ts=needs_ts)
-            out.dump(f'#define {bigcount_name} P{bigcount_name}')
-        if gen_f90 == False:
-            name = util.fortran_f08_generic_interface_name(template.prototype.name)
+            names.append(util.fortran_name(template.prototype.name, bigcount=True, needs_ts=needs_ts))
```
Comment (jsquyres) [inline ompi/mpi/bindings/ompi_bindings/fortran.py]: `gen_f90` is not forwarded, so when generating the f90 (`mpi` module) outputs
`util.fortran_name()` falls back to its `gen_f90=False` branch and emits *f08* names into
f90 files. Verified in the real generated output: `mpi-ignore-tkr-interfaces-generated.h`
now contains `#define MPI_Comm_attach_buffer_c_f08ts PMPI_Comm_attach_buffer_c_f08ts`,
`#define MPI_Comm_detach_buffer_c_f08 PMPI_Comm_detach_buffer_c_f08`,
`#define MPI_Status_set_elements_c_f08 PMPI_Status_set_elements_c_f08`, etc. Those names
do not exist in the `mpi` module (both `generate_code` and `generate_interface` skip
bigcount entirely when `gen_f90`), so these are dead `#define`s that get injected into
`module mpi` by the new `OMPI_BUILD_MPI_PROFILING 1` include and stay in effect for the
rest of the file — harmless today only by luck of no name collisions. Since f90 has no
bigcount bindings at all, the right fix is to skip the bigcount name when `gen_f90` (or,
at minimum, pass `gen_f90=gen_f90` so the emitted name is not an f08 name).

File: ompi/mpi/fortran/use-mpi-ignore-tkr/Makefile.am [hunk 2]
```diff
@@ -77,6 +77,28 @@ nodist_lib@OMPI_LIBMPI_NAME@_usempi_ignore_tkr_la_SOURCES += \
         mpi-ignore-tkr-sizeof.f90
 endif
 
+api_f90_generated_libs =
+if OMPI_FORTRAN_HAVE_TS
+noinst_LTLIBRARIES = \
+        libusempi_ignore_tkr_api.la \
+        libusempi_ignore_tkr_api_profile.la
+
+nodist_libusempi_ignore_tkr_api_la_SOURCES = api_f90_generated.F90
```
Comment (jsquyres) [inline ompi/mpi/fortran/use-mpi-ignore-tkr/Makefile.am]: The two new generated sources (`api_f90_generated.F90` here and `api_f90_generated.c` in
`base/`) are declared `nodist_`, i.e. deliberately excluded from the distribution tarball,
and cleaned via `DISTCLEANFILES`. That contradicts both precedents:

* the sibling generated file *in this same Makefile.am* — `mpi-ignore-tkr-interfaces-generated.h`
  is listed in the dist `lib@OMPI_LIBMPI_NAME@_usempi_ignore_tkr_la_SOURCES` (line 66) and in
  `MAINTAINERCLEANFILES`;
* `use-mpi-f08/Makefile.am` and `use-mpi-f08/base/Makefile.am`, where `api_f08_generated.F90`
  and `api_f08_generated.c` are dist sources with `MAINTAINERCLEANFILES`.

Net effect: a tarball will ship one generated artifact of this directory but not the other
two, so tarball builds must now run the Python generator for the `.F90`/`.c` while the
`.h` is pre-shipped — an arbitrary split that will confuse the next person, and a change
in tarball build requirements that isn't called out anywhere. Either follow the f08
pattern (dist sources + `MAINTAINERCLEANFILES`) or deliberately switch all three, and note
it in the commit message. Relatedly, `MAINTAINERCLEANFILES` (line 197) was reflowed to
multi-line but `api_f90_generated.F90` was not added to it, and `base/Makefile.am` has no
`MAINTAINERCLEANFILES` at all.

File: ompi/mpi/fortran/use-mpi-ignore-tkr/Makefile.am [hunk 3]
```diff
@@ -77,6 +77,28 @@ nodist_lib@OMPI_LIBMPI_NAME@_usempi_ignore_tkr_la_SOURCES += \
...
+libusempi_ignore_tkr_api_la_FCFLAGS = \
+        $(AM_FCFLAGS) \
+        -DOMPI_BUILD_MPI_PROFILING=0
+
+nodist_libusempi_ignore_tkr_api_profile_la_SOURCES = api_f90_generated.F90
+libusempi_ignore_tkr_api_profile_la_FCFLAGS = \
+        $(AM_FCFLAGS) \
+        -DOMPI_BUILD_MPI_PROFILING=1
+
+api_f90_generated_libs += \
```
Comment (jsquyres) [inline ompi/mpi/fortran/use-mpi-ignore-tkr/Makefile.am:96]: The two new convenience libraries are only reachable through `LIBADD`, and no
`lib@OMPI_LIBMPI_NAME@_usempi_ignore_tkr_la_DEPENDENCIES` is set. Automake's `all-am`
prerequisite list is `$(LTLIBRARIES)` = `$(lib_LTLIBRARIES) $(noinst_LTLIBRARIES)`, so the
installed library is ordered *before* the convenience libraries it consumes, and
Automake's automatic LIBADD→DEPENDENCIES inference is unreliable for names hidden behind a
make variable. `use-mpi-f08/Makefile.am:396-402` hit precisely this and added an explicit
`_DEPENDENCIES` with the comment "adding it to LIBADD doesn't enforce any ordering".
Mirror that here (or list the `.la` files literally in `LIBADD`) so a serial or `-j` build
cannot attempt the link before the convenience archives exist.

**Review by jsquyres — COMMENTED** [review]
## Review Summary

**Verdict: COMMENT**

### Findings (3 validated from 12 raw)

- Agreed (cross-source): 0
- v3-reviewer: 3
- Description accuracy: 0

### Filter Pipeline

- Dropped (no quote): 0
- Dropped (hallucination): 0 (path: 0, quote: 0, deleted: 0)
- Suppressed (dedup): 1
- Dropped (below cutoff): 8

### v3-reviewer

**[Bugs and logic errors | MEDIUM]** `ompi/mpi/bindings/ompi_bindings/fortran.py:302` — _see inline comment_

**[Architecture and design patterns | MEDIUM]** `ompi/mpi/fortran/use-mpi-ignore-tkr/Makefile.am:86` — _see inline comment_

**[Bugs and logic errors | MEDIUM]** `ompi/mpi/fortran/use-mpi-ignore-tkr/Makefile.am:96` — _see inline comment_

**Comment by N00byKing** [comment]
I've modified the pympi parsing to apply to f90 as well, so now the x parameters in the interfaces are gone.
Well, more accurately *would* be gone if pympi had the definitions for {comm,session}_attach_buffer.
For that to actually happen #14216 would have to land, but it worked fine when manually updating the submodule.
Since this PR also works without the update to pympi, it doesn't block on #14216 though, and it will just work once the submodule is updated.
Regarding the comment, I'm pretty sure its just that it wasn't implemented yet, since keyword params worked in testing and the other callsite in [L396](https://github.com/open-mpi/ompi/blob/1dde44fed3686ec90e1d4fce12f5339707744de2/ompi/mpi/bindings/ompi_bindings/fortran.py#L396) just says that f90 was "out of scope".

The other review comments should be addressed as well (f08 leaking into the generated header, Makefile deps + nodist removed).

File: ompi/mpi/fortran/use-mpi-ignore-tkr/pmpi-ignore-tkr-interfaces.h [hunk 4]
```diff
@@ -264,7 +264,6 @@
 #define MPI_Status_f2f08 PMPI_Status_f2f08
 #define MPI_Status_f082f PMPI_Status_f082f
 #define MPI_Status_set_cancelled PMPI_Status_set_cancelled
```
Comment (jsquyres) [inline ompi/mpi/fortran/use-mpi-ignore-tkr/pmpi-ignore-tkr-interfaces.h:266]: The deletion of `#define MPI_Status_set_elements PMPI_Status_set_elements` from this list is correct but load-bearing and undocumented. `mpi-ignore-tkr.F90` includes this file (line 51) *before* it includes `mpi-ignore-tkr-interfaces-generated.h` (line 54), and this file's macros are never undefined — so any name that is both listed here and present in `use-mpi-ignore-tkr/Makefile.prototype_files` causes the first (non-profiling) inclusion of the generated header to emit `interface PMPI_Foo` instead of `interface MPI_Foo`, i.e. the public interface disappears from the `mpi` module with no compile-time error. That is exactly the bug this deletion fixes for `MPI_Status_set_elements`, and nothing prevents it recurring the next time a `.c.in` template is added for a function already listed here. Either add a comment at the top of this file stating the invariant, or reorder `mpi-ignore-tkr.F90` so the generated header is included before `pmpi-ignore-tkr-interfaces.h`.

File: ompi/mpi/fortran/use-mpi-ignore-tkr/mpi-ignore-tkr.F90 [hunk 5]
```diff
@@ -52,6 +52,9 @@ module mpi
 # include "ompi/mpi/fortran/use-mpi-ignore-tkr/mpi-ignore-tkr-file-interfaces.h"
 # include "ompi/mpi/fortran/use-mpi-ignore-tkr/pmpi-ignore-tkr-file-interfaces.h"
 # include "ompi/mpi/fortran/use-mpi-ignore-tkr/mpi-ignore-tkr-interfaces-generated.h"
+
+# define OMPI_BUILD_MPI_PROFILING 1
```
Comment (jsquyres) [inline ompi/mpi/fortran/use-mpi-ignore-tkr/mpi-ignore-tkr.F90]: `OMPI_BUILD_MPI_PROFILING` is defined and never `#undef`'d, and the ~25 `#define MPI_Xxx PMPI_Xxx` rename macros that the second inclusion installs (from `print_profiling_rename_macros`) also stay in effect for the remainder of the compilation unit. Everything included after line 57 — `mpi-ignore-tkr-removed-interfaces.h`, `pmpi-ignore-tkr-removed-interfaces.h` (lines 63-64) and `mpi-ignore-tkr-sizeof.h` (line 67) — is now preprocessed with those macros live. It happens to be harmless today only because none of those names overlap; the moment a name in `Makefile.prototype_files` also appears in the removed-interfaces or sizeof headers, an `MPI_` interface is silently rewritten into a `PMPI_` one with no diagnostic. Add `#undef OMPI_BUILD_MPI_PROFILING` immediately after the second include, and a comment noting that the rename macros deliberately leak (they cannot be undone) so future includes must be added *before* line 54, not after.

File: ompi/mpi/fortran/use-mpi-ignore-tkr/Makefile.am [hunk 6]
```diff
@@ -141,27 +165,45 @@ if OMPI_FORTRAN_HAVE_TS
...
 	    --srcdir $(abs_top_srcdir) \
 	    --output $(abs_builddir)/$@ \
 	    fortran \
 	    $(gen_ts) \
+        --pympistd-dir $(abs_top_srcdir)/3rd-party/pympistandard \
 	    --fort-std f90 \
 	    interface \
 	    --prototype-files $(template_files)
 
+api_f90_generated.F90: $(template_files) $(OMPI_BINDINGS_GENERATOR)
```
Comment (jsquyres) [inline ompi/mpi/fortran/use-mpi-ignore-tkr/Makefile.am:181]: `use-mpi-f08/Makefile.am` gained a `check-local` target that runs `ompi/mpi/bindings/check_f08_names.py` precisely to catch the generated mpi_f08 interfaces drifting from the pympistandard dummy-argument names. This PR extends the same standard-name substitution to the f90/`mpi`-module interfaces (`fortran.py:406`, `standard_parameter_names(..., use_f90=gen_f90)`) but adds no equivalent `check-local` for `mpi-ignore-tkr-interfaces-generated.h` / `api_f90_generated.F90`. Nor is there any `make check` test that actually calls one of the new procedures (e.g. `MPI_Comm_attach_buffer` / `PMPI_Comm_attach_buffer` from `use mpi`), so a silently-broken `_FTS` glue layer or a mismatched interface would not be caught in CI. Adding a `check-local` mirroring the f08 one (with `--fort-std f90` naming) plus a small `test/` program that calls the new generic through the `mpi` module would close the gap.

**Review by jsquyres — COMMENTED** [review]
## Review Summary

**Verdict: COMMENT**

### Findings (3 validated from 14 raw)

- Agreed (cross-source): 0
- v3-reviewer: 3
- Description accuracy: 0

### Filter Pipeline

- Dropped (no quote): 0
- Dropped (hallucination): 0 (path: 0, quote: 0, deleted: 0)
- Suppressed (dedup): 2
- Dropped (below cutoff): 9

### Warnings

- Agents failed: codex-reviewer
- Degraded — codex-reviewer: stalled (no new output within the idle window) — treated as timed out

### v3-reviewer

**[Cross-file consistency | MEDIUM]** `ompi/mpi/fortran/use-mpi-ignore-tkr/pmpi-ignore-tkr-interfaces.h:266` — _see inline comment_

**[Edge cases and error handling | MEDIUM]** `ompi/mpi/fortran/use-mpi-ignore-tkr/mpi-ignore-tkr.F90:56` — _see inline comment_

**[Test coverage | MEDIUM]** `ompi/mpi/fortran/use-mpi-ignore-tkr/Makefile.am:180` — _see inline comment_

**Comment by N00byKing** [comment]
Addressed review comments regarding the makefile.

As for the testing infra, I'm splitting that up into a separate PR.
First, this one is getting bloated (It was supposed to be a quick -1 line PR...).
More importantly, check_f08_names.py fails if it doesn't find the interface in pympistd.
If I add the check without #14216, it will fail, and thus this PR becomes dependant on #14216.
Finally, when I did modify the check, I actually did find some inconsistencies in the handwritten portion of use-mpi.
So all of that + bugfixes will go in the separate PR.

EDIT: See #14229

**Review by hppritcha — DISMISSED** [review]

File: ompi/mpi/fortran/use-mpi-ignore-tkr/base/Makefile.am [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+include ../Makefile.prototype_files
+template_files =${prototype_files:%=$(abs_top_srcdir)/ompi/mpi/fortran/use-mpi-f08/%}
+
+api_f90_generated.c: $(template_files) $(OMPI_BINDINGS_GENERATOR)
+	$(OMPI_V_GEN) $(PYTHON) $(top_srcdir)/ompi/mpi/bindings/bindings.py \
+	    --builddir $(abs_top_builddir) \
+	    --srcdir $(abs_top_srcdir) \
+	    --output $(abs_builddir)/$@ \
+	    fortran \
+	    --generate-ts-suffix \
```
Comment (jsquyres) [inline ompi/mpi/fortran/use-mpi-ignore-tkr/base/Makefile.am:44]: This rule hardcodes `--generate-ts-suffix`, while the two sibling generation rules in the parent directory (`mpi-ignore-tkr-interfaces-generated.h` and `api_f90_generated.F90` in `ompi/mpi/fortran/use-mpi-ignore-tkr/Makefile.am`) use the conditional `$(gen_ts)` (empty unless `OMPI_FORTRAN_HAVE_TS`). On a build machine whose Fortran compiler lacks TS 29113, `make dist` will therefore ship an `api_f90_generated.c` containing all the `_wrapper_f90` C wrappers alongside an `api_f90_generated.F90` that contains no procedures at all (the generator skips every non-`needs_ts` template for f90, and with `$(gen_ts)` empty nothing is `needs_ts`). The three rules should agree — either add the `if OMPI_FORTRAN_HAVE_TS` / `gen_ts` block here, or document why the C side is intentionally generated unconditionally.

**Review by jsquyres — COMMENTED** [review]
## Review Summary

**Verdict: COMMENT**

### Findings (1 validated from 17 raw)

- Agreed (cross-source): 0
- v3-reviewer: 1
- Description accuracy: 0

### Filter Pipeline

- Dropped (no quote): 0
- Dropped (hallucination): 0 (path: 0, quote: 0, deleted: 0)
- Suppressed (dedup): 0
- Dropped (below cutoff): 16

### v3-reviewer

**[cross-file-consistency | MEDIUM]** `ompi/mpi/fortran/use-mpi-ignore-tkr/base/Makefile.am:44` — _see inline comment_

File: ompi/mpi/fortran/use-mpi-ignore-tkr/base/Makefile.am (hunk 7 above)
Comment (N00byKing) [reply ompi/mpi/fortran/use-mpi-ignore-tkr/base/Makefile.am:44]: Automake is weird. The entire file was already guarded by OMPI_FORTRAN_HAVE_TS, but this still makes a difference...
Anyway, fixed now.

File: ompi/mpi/fortran/use-mpi-ignore-tkr/Makefile.am [hunk 8]
```diff
@@ -87,8 +109,10 @@ endif
 # directly (pulling it in indirectly via libmpi.la does not work on
 # all platforms).
 lib@OMPI_LIBMPI_NAME@_usempi_ignore_tkr_la_LIBADD = \
+        $(api_f90_generated_libs) \
```
Comment (jsquyres) [inline ompi/mpi/fortran/use-mpi-ignore-tkr/Makefile.am:112]: This PR puts C code into `libmpi_usempi_ignore_tkr` for the first time (the TS choice-buffer wrappers in `base/libusempi_ignore_tkr_ccode.la`), and that C calls into libmpi: `PMPI_Comm_f2c`, `PMPI_Comm_attach_buffer`, `PMPI_Session_attach_buffer`, and the errhandler machinery behind `OMPI_ERRHANDLER_INVOKE`. But this `LIBADD` only pulls in OPAL.

The sibling Fortran libraries that contain C code link libmpi explicitly for exactly this reason — see `ompi/mpi/fortran/mpif-h/Makefile.am` (whose comment notes that pulling libraries in indirectly "does not work on all platforms") and `ompi/mpi/fortran/use-mpi-f08/Makefile.am`. On ELF/macOS this happens to work only because the mpifort wrapper also links `-lmpi`, but it leaves undefined symbols in the shared library, and on platforms/link modes that require fully-resolved shared objects the link will fail outright.

Could you add `$(top_builddir)/ompi/lib@OMPI_LIBMPI_NAME@.la` to this `LIBADD`, matching mpif-h and use-mpi-f08?

File: ompi/mpi/fortran/use-mpi-ignore-tkr/Makefile.am [hunk 9]
```diff
@@ -87,8 +109,10 @@ endif
...
         $(libusempi_internal_modules) \
         $(OMPI_TOP_BUILDDIR)/opal/lib@OPAL_LIB_NAME@.la
+lib@OMPI_LIBMPI_NAME@_usempi_ignore_tkr_la_DEPENDENCIES = $(api_f90_generated_libs)
```
Comment (jsquyres) [inline ompi/mpi/fortran/use-mpi-ignore-tkr/Makefile.am]: Setting `_DEPENDENCIES` explicitly replaces the default dependency list that Automake derives from `_LIBADD`, so the installed library loses its make-level dependencies on `$(libusempi_internal_modules)` and the OPAL library — rebuilding those will no longer trigger a relink of `libmpi_usempi_ignore_tkr`, which can leave stale incremental builds. (SUBDIRS ordering keeps a from-scratch build safe, and use-mpi-f08 uses the same idiom, so this is an inherited pattern — but it's worth not propagating it.)

Suggest either dropping this override (Automake's derived list should already cover `$(api_f90_generated_libs)` since it appears in `LIBADD`), using `EXTRA_lib@OMPI_LIBMPI_NAME@_usempi_ignore_tkr_la_DEPENDENCIES`, or listing the full `LIBADD` set here.

**Review by jsquyres — COMMENTED** [review]
First off: apologies for the slow turnaround on this one, and thank you for your patience — I know this PR has been through several rounds already, and I appreciate you sticking with it. I'm sorry to come back with yet more build-system feedback; the ignore-TKR + TS plumbing is a genuinely subtle corner of the tree.

The good news: the core fix (removing the stray `MPI_Status_set_elements` PMPI define) looks correct to me — it was the only overlap between the PMPI define list and the generated prototypes — and the double-include profiling scheme holds together.

I've left three inline comments below: one linking concern and two Automake dependency/distribution concerns.

File: ompi/mpi/fortran/use-mpi-ignore-tkr/base/Makefile.am [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+# The one .c file here backs the mpi module's TS29113 choice-buffer
+# procedures.
+# This directory + Makefile.am is separate to have non-zeroed CPPFLAGS/AM_CPPFLAGS
+if OMPI_FORTRAN_HAVE_TS
+
+noinst_LTLIBRARIES = libusempi_ignore_tkr_ccode.la
+
+libusempi_ignore_tkr_ccode_la_SOURCES = api_f90_generated.c
+
+gen_ts = --generate-ts-suffix
```
Comment (jsquyres) [inline ompi/mpi/fortran/use-mpi-ignore-tkr/base/Makefile.am]: Following on from the earlier guard discussion (sorry to keep poking at this corner): `gen_ts` is only set inside `if OMPI_FORTRAN_HAVE_TS`, but the `api_f90_generated.c` rule below sits *outside* that conditional — and `api_f90_generated.c` (plus the generated `.F90` and interface header in the parent directory, which follow the same pattern) are distributed sources.

Consequence: running `make dist` on a machine whose Fortran compiler lacks TS 29113 invokes the generator with an empty `$(gen_ts)`; the generator's new `if gen_f90 and not needs_ts: continue` then skips every template, so the tarball ships essentially empty generated files — silently breaking `use mpi` TS support for anyone building that tarball with a TS-capable compiler.

Since *compilation* is already gated by `OMPI_FORTRAN_HAVE_TS`, I believe the generation rules could pass `--generate-ts-suffix` unconditionally so dist artifacts are always complete.

Comment (N00byKing) [reply ompi/mpi/fortran/use-mpi-ignore-tkr/base/Makefile.am]: I'll do the other comments tomorrow, but I dont understand this one.
Wasn't the issue last time that you wanted it to be conditional on HAVE_TS?
The previous version did hardcode the flag after all

File: ompi/mpi/fortran/use-mpi-ignore-tkr/Makefile.am (hunk 9 above)
Comment (N00byKing) [reply ompi/mpi/fortran/use-mpi-ignore-tkr/Makefile.am]: I dont think dropping it makes sense considering you told be to add it two rounds earlier...
Added the EXTRA_ prefix

File: ompi/mpi/fortran/use-mpi-ignore-tkr/base/Makefile.am (hunk 10 above)
Comment (N00byKing) [reply ompi/mpi/fortran/use-mpi-ignore-tkr/base/Makefile.am]: Followup: I just hardcoded it again.

**Review by hppritcha — APPROVED** [review]

**Comment by hppritcha** [comment]
@N00byKing would you mind squashing down to one commit?  I'll rereview and merge after that.

**Comment by N00byKing** [comment]
Done

**Comment by jsquyres** [comment]
Thank you @N00byKing!
